src/main/java/net/dries007/tfc/objects/blocks/metal/BlockMetalLamp.java

```diff
@@ -0,0 +1,514 @@
+import net.dries007.tfc.objects.blocks.property.ILightableBlock;
+import net.dries007.tfc.objects.te.TELamp;
+import net.dries007.tfc.util.Helpers;
+import net.dries007.tfc.util.OreDictionaryHelper;
+import net.dries007.tfc.util.calendar.ICalendar;
+
+@ParametersAreNonnullByDefault
+public class BlockMetalLamp extends Block implements ILightableBlock
+{
+
+    private static final Map<Metal, BlockMetalLamp> MAP = new HashMap<>();
+    private static final AxisAlignedBB AABB_UP = new AxisAlignedBB(0.3125, 0, 0.3125, 0.6875, 0.5, 0.6875);
+    private static final AxisAlignedBB AABB_DOWN = new AxisAlignedBB(0.3125, 0, 0.3125, 0.6875, 1, 0.6875);
+
+    private final Metal metal;
+
+    public static final PropertyDirection FACING = PropertyDirection.create("facing", new Predicate<EnumFacing>()
+    {
+        public boolean apply(@Nullable EnumFacing p_apply_1_)
+        {
+            return p_apply_1_ == EnumFacing.DOWN || p_apply_1_ == EnumFacing.UP;
+        }
+    });
+
+    public BlockMetalLamp(Metal metal)
+    {
+        super(Material.IRON);
+        this.metal = metal;
+        setDefaultState(blockState.getBaseState().withProperty(FACING, EnumFacing.UP).withProperty(LIT, false));
+        setHardness(1f);
+        setLightLevel(0.9375F);
+        setTickRandomly(true);
+        setSoundType(SoundType.METAL);
+        if (!MAP.containsKey(metal))
+            MAP.put(metal, this);
+
+        OreDictionaryHelper.register(this, "lamp");
+    }
+
+    public static BlockMetalLamp get(Metal metal)
+    {
+        return MAP.get(metal);
+    }
+
+    public static ItemStack get(Metal metal, int amount)
+    {
+        return new ItemStack(MAP.get(metal), amount);
+    }
+
+    public Metal getMetal()
+    {
+        return metal;
+    }
+
+    //Don't need to do any clientside display ticking, as no smoke particles... yet
+    //public void randomDisplayTick(IBlockState stateIn, World worldIn, BlockPos pos, Random rand)
+
+    //may support wall attachments sometime
+    @SuppressWarnings("deprecation")
+    @Override
+    @Nonnull
+    public IBlockState getStateFromMeta(int meta)
+    {
+        IBlockState iblockstate = this.getDefaultState();
+
+        switch (meta)
+        {
+            case 0:
+                iblockstate = iblockstate.withProperty(FACING, EnumFacing.DOWN);
+                break;
+            case 1:
+                iblockstate = iblockstate.withProperty(FACING, EnumFacing.EAST);
+                break;
+            case 2:
+                iblockstate = iblockstate.withProperty(FACING, EnumFacing.WEST);
+                break;
+            case 3:
+                iblockstate = iblockstate.withProperty(FACING, EnumFacing.SOUTH);
+                break;
+            case 4:
+                iblockstate = iblockstate.withProperty(FACING, EnumFacing.NORTH);
+                break;
+            case 5:
+            default:
+                iblockstate = iblockstate.withProperty(FACING, EnumFacing.UP);
```

Comment:
This is both redundant (as the facing is restricted to only UP / DOWN), and overly verbose. For example:
```java
return state.withProperty(FACING, EnumFacing.byIndex(meta % 2)).withProperty(LIT, meta >= 2);
```